### PR TITLE
select menus

### PR DESCRIPTION
**Pull request by TopiSenpai**
https://discord.com/developers/docs/interactions/message-components#select-menus

### CODE REVIEW

**Comment by TopiSenpai** [comment]
they are pretty much ready
maybe I add some convenient methods on the Button/Select Menu Events to get the used component from the message idk

**Comment by EveCodes31** [comment]
Is this meant to be based on master @TopiSenpai?

File: internal/handlers/guild_delete_handler.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (h GuildDeleteHandler) HandleGatewayEvent(disgo api.Disgo, eventManager api
 	eventManager.Dispatch(genericGuildEvent)
 
 	if guild.Unavailable {
-		// set guild to unavail for now
+		// set guild to unavailable for now
 		disgo.Cache().Guild(guild.ID).Unavailable = true
```
Comment (EveCodes31) [inline internal/handlers/guild_delete_handler.go:38]: Couldn't this cause an NPE?

Comment (TopiSenpai) [reply internal/handlers/guild_delete_handler.go]: yes good catch

**Comment by TopiSenpai** [comment]
> Is this meant to be based on master @TopiSenpai?

fixed 👀

**Review by EveCodes31 — APPROVED** [review]

**Pull request merged**